Surname by branch

lledieu opened the pull request:
Rework surname_by_branch / m=N;v=

This page wasn't compliant with HTML5 : 
- dl/dt/dd only used on first level
- ul/li + CSS for others
- added class posterity to be able to inline "..."

lledieu committed:
Rework surname_by_branch / m=N;v=

This page wasn't compliant with HTML5.
- dl/dt/dd only used on first level
- ul/li + CSS for others
- added class posterity to be able to inline "..."

Signed-off-by: Ludovic Ledieu <lledieu@free.fr>

lledieu commented:
À noter qu'il est également possible de simplifier le code en gérant le toggle au niveau javascript... Si ça vous intéresse, je peux l'ajouter.

```
$(document).ready(function() {
  $(document).on( "click", "#surname_by_branch .toggle", function(e) {
    if( $(this).text() == "o" ) {
      $(this).text( "+" );
      $(this).closest( "li" ).children( "ul" ).hide();
    } else {
      $(this).text( "o" );
      $(this).closest( "li" ).children( "ul" ).show();
    }
  });
});
```

lledieu committed:
Fixed wrong link for spouses

Signed-off-by: Ludovic Ledieu <lledieu@free.fr>

a2line commented:
Joli travail ! Je me dis que le javascript est bon à prendre s'il masque le bloc sur place, ça éviterait le rafraîchissement de la page et la requête serveur supplémentaire sous-jacente (l'ancre cliquée remonte en haut de l'écran, sous Chrome ça passe en plus par le haut de page avant de revenir à l’ancre… problème qu'on a d’ailleurs sur les dicos où les pages sont bien plus lourdes).

Utiliser l’ellipse « … » (élision) pour marquer la postérité, plutôt que les trois petits points « ... ».

sagotch commented:
Je viens de relancer le test travis qui devrait passer (c'était juste un timeout sous macos). Pas encore testé mais ça me semble ok :+1:

lledieu commented:
@a2line J'ai essayé de trouver des symbols unicode pour remplacer les puces. J'ai trouvé ça mais je me demande si ça convient : 
&#x229D; &#x2295; &#x2299;
As-tu d'autres idées ? Peut-être qu'en utilisant la taille small, ça me choque moins...

Si on utilise la version javascript, je vais faire disparaître le paramètre u=. Est-ce acceptable ?

lledieu commented:
Pour expliciter un peu plus le paramètre u= qui disparaît, si j'ajoute à l'url &cgl=on, je retrouve l'état initial ce qui pourrait être perçu comme une régression...

a2line commented:
Oui forcément en javascript on a les données au complet sans u=. Ça peut être vu comme regression si on veut partager un arbre réduit qui ne contient pas les informations dans la source.

⊝⊜⊕⊙⊗  ◴◵◶◷  ⧁☉⦿⚪⚇⚆⚫⚉⚈  ⚣⚤

a2line commented:
L’un des avantages des u= en URL est qu'on peut savoir ce qu'on a réduit et en refaire une liste en haut d’arborescence pour les retrouver plus facilement avec des ancres ou boutons accolés les uns aux autres pour les développer à nouveau. On les perd vite de vu !

lledieu committed:
Conflict was possible on id

bullet id are now as id=if<ifam> to make sure there is no conflict with
ids from reference id=i<iper>

Signed-off-by: Ludovic Ledieu <lledieu@free.fr>

lledieu commented:
Pour la version javascript, j'ai préféré créer une autre branche expérimentale car je pense qu'il faudra un peu plus de temps pour mettre au point : 
https://github.com/geneweb/geneweb/compare/master...lledieu:surname_by_branch_v2

hgouraud commented:
For the two top buttons (Toggle/untoggle al), use the same (large) icon as in the body.
I dont get the purpose of the "fix toggle in url" button. It adds an extra &= in the url at each click

lledieu commented:
> For the two top buttons (Toggle/untoggle al), use the same (large) icon as in the body.

Puisque la v2 est une branche expérimentale, je n'en suis pas encore dans les finitions. Cependant peux-tu préciser ta pensée que je puisse comprendre ? En gros qu'est ce que j'aurais dû mettre...

> I dont get the purpose of the "fix toggle in url" button. It adds an extra &= in the url at each click

Effectivement, il faut que je revois la construction de la requête...

Pour ce qui est de l'intérêt du "fix toggle in url", il suffit de lire les commentaires ci-dessus.

> Si on utilise la version javascript, je vais faire disparaître le paramètre u=. Est-ce acceptable ?
Pour expliciter un peu plus le paramètre u= qui disparaît, si j'ajoute à l'url &cgl=on, je retrouve l'état initial ce qui pourrait être perçu comme une régression...

En gros, quand tu masques des éléments de branches en v2, tout se fait à l'écran. Si tu veux stabiliser tes actions dans l'URL, il faut utiliser ce bouton. Si tu utilises "toggle all" avant, tu comprendra mieux...

lledieu commented:
Il y a une régression sur cette branche que j'ai pu confirmer en testant une autre PR. Quand les époux ont même patronyme, il n'est pas affiché sur l'époux/épouse => j'ai dois donc corriger ça !

a2line commented:
Il y a aussi un rebase à faire sur le master avec deux conflits à résoudre.

a2line commented:
Et décider quelle version on veut, on masque en js ou non ? Ce choix est assez embêtant car les deux ont des avantages/inconvénients…

lledieu committed:
Adjusments for wrong rebase + correct missing surname for spouse

Signed-off-by: Ludovic Ledieu <lledieu@free.fr>

lledieu commented:
Rebase fait avec ajustement du problème signalé hier.
Je pense que la v2 est encore expérimentale (branche hors PR). Il existe un plugin webtrees qui semble faire un truc similaire et qu'il serait intéressant d'étudier pour y piocher d'éventuelles bonnes idées.

lledieu commented:
I think this PR is now ready for merge...

sagotch commented:
:+1: Je regarderais ça en meme temps que https://github.com/geneweb/geneweb/pull/842

hgouraud commented on lib/util.ml:
```diff
@@ -1004,14 +1004,18 @@ let geneweb_link conf href s =
-      ["<a href=\""; commd conf; acces conf base p; "\" id=\"i";
-       string_of_iper iper; "\">"; s; "</a>"]
+      ["<a href=\""; commd conf; acces conf base p;
+       if with_id then string_of_iper iper else "";
```
yes

lledieu replied:
Mon dieu, c'est vieux ! Faut que je retrouve pourquoi j'ai fait ce changement...

sagotch replied:
Oui, bon anniversaire en retard à la PR du coup :birthday:

lledieu replied:
Dans ma version locale qui fonctionne pour moi depuis 1 an j'ai : 
```
let reference_flags with_id conf base p s =
  let iper = get_key_index p in
  if conf.cancel_links || is_hidden p then s
  else
    String.concat ""
      ["<a href=\""; commd conf; acces conf base p;
       if with_id then "\" id=\"i" else "";
       if with_id then string_of_int (Adef.int_of_iper iper) else "";
       "\">"; s; "</a>"]
```

sagotch replied:
À remettre donc, ok.

sagotch replied:
On  dirait bien.

sagotch commented on lib/some.ml:
```diff
@@ -429,7 +429,7 @@ let print_branch conf base psn name =
       | _ -> None
     in
-    let print_elem p with_link with_id =
+    let print_elem p with_link with_id with_sn =
```
What is the difference between `with_sn` and `psn`? I guess that `psn` stand for "print surname", so I do not think there is a need for a new argument here.

hgouraud replied:
probable, mais il vaudrait mieux valider avec @lledieu

lledieu replied:
D'après le commentaire du commit, j'ai été obligé de rajouter ça pour gérer le cas de figure où le nom des 2 époux est le même ! psn est un paramètre utilisateur pour conditionner l'affichage, with_sn est un paramètre technique pour forcer l'affichage du nom quand on affiche l'époux... Donc ce n'est pas la même chose.

sagotch replied:
Au temps pour moi j'ai vu 3 boolean a chaque fois et j'ai cru qu'il y en avais un pour psn mais non

lledieu committed:
A part was missing

Signed-off-by: Ludovic Ledieu <lledieu@free.fr>

lledieu commented:
J'ai fait un rebase sans trop comprendre donc j'espère avoir fait ce qu'il fallait...

sagotch commented:
Alors visiblement pas, il y a plein de commits et impossible de rebase et merge depuis github.

> This branch cannot be rebased due to conflicts
Rebasing the commits of this branch on top of the base branch cannot be performed automatically due to conflicts encountered while reapplying the individual commits from the head branch.

Je suppose que tu as voulu utiliser le rebase depuis l'interface github. Je ne sais pas comment ça marche mais pour avoir essayé ça ne donne pas le résultat esperé.

Peux-tu rebase à la main (au pire cherry-pick vu le nombre de commits). Et je pense que tu peux tout squash/fixup dans un seul commit.

lledieu commented:
C'est bien l'impression que j'avais :( Je vais faire une nouvelle branche plus propre...

lledieu commented:
Je pense avoir réparé le rebase...

sagotch commented:
Ça me semble bon maintenant !

Par contre pour cette histoire de `⊝ ⊕ ⊙` effectivement je trouve ça mieux que l'actuel.

Si ça te va, je le ajouterai et je vais *squash* les commis en un seul avant de merger.

lledieu commented:
Comme tu veux.

sagotch commented:
Mergé à la main, merci https://github.com/geneweb/geneweb/commit/d97a43de03cfe1c6127104927583f248caea74e0

(je n'ai pas remplacé les bullets car je ne suis en fait pas convaincu du resultat si la police est trop petite)
